src/Commerce/Generator/PayPal/CLI.php

```diff
@@ -0,0 +1,480 @@
+		$order_status = $this->parse_order_status( $assoc_args );
+		$post         = get_post( $post_id );
+
+		WP_CLI::log( "Generating PayPal {$orders_count} orders for post {$post_id}" );
+
+		$progress  = make_progress_bar( 'Generating orders', $orders_count );
+		$generated = array();
+
+		$this->hijack_request_flow();
+
+		for ( $k = 0; $k < $orders_count; $k ++ ) {
+			$user_id   = 0;
+			$ticket_id = $ticket_ids[ array_rand( $ticket_ids, 1 ) ];
+			$ticket    = $paypal->get_ticket( $post_id, $ticket_id );
+
+			$this->backup_ticket_total_sales( $ticket_id );
+
+			$ticket_qty = mt_rand( $tickets_min, $tickets_max );
+
+			$inventory  = $ticket->inventory();
+			$inventory  = - 1 === $inventory ? 'unlimited' : $inventory;
+			$ticket_qty = 'unlimited' === $inventory ? $ticket_qty : min( $ticket_qty, (int) $inventory );
+
+			WP_CLI::log( "Generating an order for {$ticket_qty} attendees for ticket {$ticket_id}, current ticket inventory is {$inventory}" );
+
+			if ( $ticket_qty === 0 ) {
+				WP_CLI::log( "Not generating attendees for ticket {$ticket_id} as it ran out of inventory" );
+				continue;
+			}
+
+			$this->order_status = $order_status;
+
+			$progress->tick();
+
+			$faker = Faker\Factory::create();
+
+			$transaction_id = strtoupper( substr( md5( $faker->sentence ), 0, 17 ) );
+			$receiver_id    = strtoupper( substr( md5( $faker->sentence ), 0, 13 ) );
+			$payer_id       = strtoupper( substr( md5( $faker->sentence ), 0, 13 ) );
+			$ipn_track_id   = substr( md5( $faker->sentence ), 0, 13 );
+
+			$ticket_price   = $ticket->price;
+			$payment_gross  = $ticket_price * $ticket_qty;
+			$receiver_email = 'merchant@' . parse_url( home_url(), PHP_URL_HOST );
+			$payment_date   = $faker->date( 'H:i:s M d, Y e' );
+
+			$data = array(
+				'last_name'              => $faker->lastName,
+				'shipping_method'        => 'Default',
+				'address_state'          => 'CA',
+				'receiver_email'         => $receiver_email,
+				'custom'                 => '{"user_id":' . $user_id . ',"tribe_handler":"tpp"}',
+				'item_name1'             => "{$ticket->name} - {$post->post_title}",
+				'shipping_discount'      => '0.00',
+				'receiver_id'            => $receiver_id,
+				'charset'                => 'windows-1252',
+				'payer_email'            => $faker->email,
+				'protection_eligibility' => 'Eligible',
+				'item_number1'           => "{$post_id}:{$ticket_id}",
+				'address_zip'            => '95131',
+				'mc_handling1'           => '0.00',
+				'payment_fee'            => $this->signed_value( 0.09 ),
+				'transaction_subject'    => '',
+				'txn_id'                 => $transaction_id,
+				'residence_country'      => 'US',
+				'payment_status'         => ucwords( $order_status ),
+				'mc_fee'                 => $this->signed_value( 0.09 ),
+				'mc_gross'               => $this->signed_value( $payment_gross ),
+				'insurance_amount'       => '0.00',
+				'address_country'        => 'United States',
```

Comment:
Should we randomize the address in general?

Reply:
Good idea, still in the US for the time being.